### PR TITLE
Android tracking when app is minimised

### PR DESCRIPTION
**Pull request by tomasMizera**
### Makes tracking working in the background (app runs, but not visible) on the Android platform

Uses a file to pass position updates from `TrackingService (java, different thread - foreground service)` to `AndroidTrackingBackend (cpp)`. In case the main activity (cpp part) is stopped when in the background, the tracking service (java) still operates and keeps writing updates to the file. Once cpp is started again it reads the missed position updates from the file and continues to show the tracking.

image

This is how we check if the tracking service is running when the app (cpp) is maximised again:

image

---

 - _Some tweakings are still to be done with tracking, but will be tackled in the followup PRs_

### CODE REVIEW

File: app/position/tracking/positiontrackingmanager.h [hunk 1]
```diff
@@ -69,14 +69,15 @@ class PositionTrackingManager : public QObject
...
+    Q_INVOKABLE void commitTrackedPath();
 
     QDateTime startTime() const;
 
     bool isTrackingPosition() const;
 
   public slots:
-    void addPoint( const GeoPosition &position );
+    void addPoint( const QgsPoint &position );
+    void addPoints( QList<QgsPoint> positions );
```
Comment (PeterPetrik) [inline app/position/tracking/positiontrackingmanager.h:80]: `const QList< QgsPoint > &`

File: app/inpututils.cpp [hunk 2]
```diff
@@ -2013,3 +2013,33 @@ QString InputUtils::imageGalleryLocation()
 
   return galleryPaths.last();
 }
+
+QList<QgsPoint> InputUtils::parsePositionUpdates( const QString &data )
+{
+  QList<QgsPoint> parsedUpdates;
```
Comment (PeterPetrik) [inline app/inpututils.cpp:2019]: Any reason why QList and not QVector is used?

**Review by wonder-sk — APPROVED** [review]
:clap: :clap: :clap:

File: app/inpututils.cpp (hunk 2 above)
Comment (tomasMizera) [reply app/inpututils.cpp:2019]: I think QVector is just an alias for QList in Qt6 
image

Comment (tomasMizera) [reply app/inpututils.cpp:2019]: https://doc.qt.io/qt-6/qvector.html

Comment (PeterPetrik) [reply app/inpututils.cpp:2019]: good to know. thanks

**Pull request merged**